### ISSUE TITLE
keys / files with spaces are not processed correctly

### ISSUE THREAD

**Issue by lhriley** [larrabee/s3sync#9]
The awscli s3 command lets a user put / get objects with spaces in the name. This is unfortunate, since it allows badly named files to be put into S3 simply by quoting the name.

For example:
```
aws s3 ls 's3://some-bucket/prefix/this file has spaces in the name(1).pdf'
```
`s3sync` does not handle this situation well, and exits with errors:

```
DEBU[0002] Putting obj prefix/this file has spaces in the name(1).pdf failed with err: mkdir /path/to/prefix/this: not a directory 
```

Since we have a collection of arbitrarily named files, and come from sources outside our control, we can't guarantee that files we wish to copy do not contain spaces or other problematic characters in the names. I haven't had an opportunity to test it, but I would suspect that this problem would happen when putting / getting from local or s3 alike.

**Comment by larrabee** [comment]
Hello.  
Did you test this on latest version. I cannot reproduce it with latest release.

**Comment by lhriley** [comment]
I have only used the code in `master` and this release: https://github.com/larrabee/s3sync/releases/tag/1.14

There might be some nuance that I didn't include in my examples.

There are spaces in both the prefixes and the object keys which, on a Linux system are not automatically escaped. This means that they are considered arguments instead of part of the path string when writing to a local filesystem.

I am running Linux Mint 19.1 (Ubuntu 18.04 Bionic), and believe I'm using `ext4`. I don't think that the filesystem should matter all that much since go should be handling that part. YMMV on Windows / MacOS.

**Comment by larrabee** [comment]
Hello again.  
Can you provide full cmdline (without S3 keys of course)?
Can you check, that file `/path/to/prefix/this` does not exist on FS?

**Comment by lhriley** [comment]
Unfortunately I haven't had an opportunity to circle back to gather more information. Since this was giving me trouble, I moved to another project for the work I needed to accomplish.

I can guarantee that there were no files on disk at the time of the errors. In fact, I `rm -rf`'d the whole tree before resyncing to verify.

I don't recall the specific flags, but I wasn't doing anything fancy. Just tried to sync s3 to disk.

Sorry that I don't have better details at the moment.
